### PR TITLE
Update aimec to compare sims from different comMods

### PR DESCRIPTION
**Pull request by awirb**
* add writing of mass_simName.txt file for com1DFAPy 
* save entrained mass from com1DFAPy
* export peakFiles for final time step to Outputs/com1DFAPy/peakFiles and other time steps to subfolder timeSteps within peakFiles
* add option to compare only two simulations, so matching simulations from two modules 
* add cleaning of avaDir from all ana3Aimec files

### CODE REVIEW

File: avaframe/ana3AIMEC/dfa2Aimec.py [hunk 1]
```diff
@@ -100,6 +104,24 @@ def extractMBInfo(avaDir):
             log.info('Saved to dfa_mass_balance/%s ' % (os.path.basename(savename)))
             countFile = countFile + 1
 
+    return countFile
+
+def getMBInfo(avaDir, countStart=0):
+    """ Get MB info """
+
+    # Get info from ExpLog
+    mbFiles = glob.glob(os.path.join(avaDir, 'Outputs', 'com1DFAPy', 'mass*.txt'))
```
Comment (fso42) [inline avaframe/ana3AIMEC/dfa2Aimec.py:113]: Could we use the same strategy as with eg. the cfgUtils and use the modules name to get the correct folder -> would make the function more generic. 
I am aware oft the problem with com1DFAPy also being called com1DFA, but we could put this behind a TODO/Dev flag and remove it when we move com1DFAPy...

File: avaframe/ana3AIMEC/dfa2Aimec.py [hunk 2]
```diff
@@ -100,6 +104,24 @@ def extractMBInfo(avaDir):
...
+    mbNames = sorted(set(mbFiles), key=lambda s: (s.split("_")[1], s.split("_")[2], s.split("_")[4]))
+
+    countFile = countStart
+    for mFile in mbNames:
+        savename = os.path.join(avaDir, 'Work', 'ana3AIMEC', 'com1DFA', 'dfa_mass_balance', '%06d.txt' % (countFile + 1))
```
Comment (fso42) [inline avaframe/ana3AIMEC/dfa2Aimec.py:118]:
```suggestion
        saveName = os.path.join(avaDir, 'Work', 'ana3AIMEC', 'com1DFA', 'dfa_mass_balance', '%06d.txt' % (countFile + 1))
```

File: avaframe/ana3AIMEC/dfa2Aimec.py [hunk 3]
```diff
@@ -117,13 +139,22 @@ def mainDfa2Aimec(avaDir, cfgSetup):
...
-        fU.getDFAData(avaDir, workDir, suf, dir)
-
     # Write the paths to this files to a file
     writeAimecPathsFile(cfgSetup, avaDir)
 
-    # Extract the MB info
-    extractMBInfo(avaDir)
+    comModules = cfgSetup['comModules'].split('|')
+    countStart = 0
+    countFile = 0
```
Comment (fso42) [inline avaframe/ana3AIMEC/dfa2Aimec.py:147]: What is the purpose of the counting done here?

File: avaframe/com1DFAPy/com1DFA.py [hunk 4]
```diff
@@ -111,7 +111,7 @@ def com1DFAMain(cfg, avaDir, relTh):
             relFiles = releaseLine['file']
             # ------------------------
             #  Start time step computation
-            Tsave, T, U, Z, S, Particles, Fields, Tcpu = DFAIterate(cfgGen, particles, fields, dem)
+            Tsave, T, U, Z, S, Particles, Fields, Tcpu = DFAIterate(cfgGen, particles, fields, dem, avalancheDir, logName)
```
Comment (fso42) [inline avaframe/com1DFAPy/com1DFA.py:114]: Maybe we should use a general `info` dict that is being returned, with eg. the Tcpu, the additional mass info + any other (potential)info in it. -> Then we could drop the avalanchDir and logName input...

File: avaframe/com1DFAPy/com1DFA.py [hunk 5]
```diff
@@ -1569,6 +1579,9 @@ def exportFields(cfgGen, Tsave, Fields, relFile, demOri, outDir, logName):
     resTypesString = cfgGen['resType']
     resTypes = resTypesString.split('_')
     tSteps = fU.getTimeIndex(cfgGen, Fields)
+    if -1 not in tSteps:
+        tSteps.append(-1)
+        log.info('-1 added to tStep')
```
Comment (fso42) [inline avaframe/com1DFAPy/com1DFA.py:1584]: What is this for?

File: avaframe/in3Utils/fileHandlerUtils.py [hunk 6]
```diff
@@ -204,7 +204,7 @@ def getTimeIndex(cfg, Fields):
     return tStepsOk
 
 
-def getDFAData(avaDir, workDir, suffix, comModule='com1DFA', nameDir=''):
+def getDFAData(avaDir, workDir, suffix, comModule='com1DFA', nameDir='', countStart=0):
```
Comment (fso42) [inline avaframe/in3Utils/fileHandlerUtils.py:207]: Same question as before, what is the countStart needed for?

File: avaframe/in3Utils/fileHandlerUtils.py [hunk 7]
```diff
@@ -226,8 +226,8 @@ def getDFAData(avaDir, workDir, suffix, comModule='com1DFA', nameDir=''):
     # Lead all infos on simulations
     inputDir = os.path.join(avaDir, 'Outputs', comModule, 'peakFiles')
     data = makeSimDict(inputDir)
-
-    countsuf = 0
+    nFiles = len(data['files'])
+    countsuf = countStart
```
Comment (fso42) [inline avaframe/in3Utils/fileHandlerUtils.py:230]:
```suggestion
    countSuf = countStart
```

**Review by fso42 — CHANGES_REQUESTED** [review]

File: avaframe/ana3AIMEC/dfa2Aimec.py (hunk 3 above)
Comment (awirb) [reply avaframe/ana3AIMEC/dfa2Aimec.py:147]: this is required as the files for aimec are renamed to 000001.txt, 000002.txt, etc and data from two modules is compared the function needs to know where to start the counter

File: avaframe/com1DFAPy/com1DFA.py (hunk 4 above)
Comment (awirb) [reply avaframe/com1DFAPy/com1DFA.py:114]: Particles, the list of particles dictionaries contains all the mass info, however only for the saving time steps, however aimec requires the mass balance info to be listed for every computation time step. For this reason, it is written to file within the function DFAiterate

File: avaframe/com1DFAPy/com1DFA.py (hunk 5 above)
Comment (awirb) [reply avaframe/com1DFAPy/com1DFA.py:1584]: that in any case, so whatever inputs you provide in the ini file for tSteps (the time steps that you want data to be exported to Outputs), the last time step is exported

File: avaframe/in3Utils/fileHandlerUtils.py (hunk 6 above)
Comment (awirb) [reply avaframe/in3Utils/fileHandlerUtils.py:207]: see answer from above

File: avaframe/in3Utils/fileHandlerUtils.py (hunk 7 above)
Comment (awirb) [reply avaframe/in3Utils/fileHandlerUtils.py]: done

File: avaframe/ana3AIMEC/dfa2Aimec.py (hunk 2 above)
Comment (awirb) [reply avaframe/ana3AIMEC/dfa2Aimec.py]: done

**Review by matthiasto — COMMENTED** [review]
We should maybe make sure to clean the aimec output and work dir before running it, otherwise there are some interference with previous calls to AIMEC.
I do not know if you want to include this in this PR or the next one

**Comment by awirb** [comment]
I included a call to clean the avaDir from all ana3Aimec module files

**Review by matthiasto — APPROVED** [review]

File: avaframe/ana3AIMEC/dfa2Aimec.py (hunk 3 above)
Comment (awirb) [reply avaframe/ana3AIMEC/dfa2Aimec.py]: removed

File: avaframe/com1DFAPy/com1DFA.py (hunk 4 above)
Comment (awirb) [reply avaframe/com1DFAPy/com1DFA.py]: changed

File: avaframe/in3Utils/fileHandlerUtils.py (hunk 6 above)
Comment (awirb) [reply avaframe/in3Utils/fileHandlerUtils.py]: removed

**Review by fso42 — APPROVED** [review]

**Pull request merged**